app/databases.py

```diff
@@ -0,0 +1,153 @@
+import os
+import logging
+import pyodbc
+from typing import Any
+from datetime import datetime, timezone
+from azure.cosmos import CosmosClient
+from azure.identity import DefaultAzureCredential
+import struct
+
+environment = os.environ.get("ENVIRONMENT", default="dev")
+
+def db_aad_token_struct() -> bytes:
+    """
+    Uses AzureCli login state to get a token for the database scope
+    Used when running 'make serve-dev'.
+
+    Kindly leveraged from this SO answer: https://stackoverflow.com/a/67692382
+    """
+    credential = DefaultAzureCredential()
+    token = credential.get_token("https://database.windows.net/")[0]
+
+    token_bytes = bytes(token, "UTF-16 LE")
+
+    return struct.pack("=i", len(token_bytes)) + token_bytes
+
+def odbc_cursor() -> Any:
+    """
+    ODBC cursor for running queries against the MSSQL feature store.
+
+    Documentation: https://github.com/mkleehammer/pyodbc/wiki
+    """
+
+    connection_string = os.environ["FEATURE_STORE_CONNECTION_STRING"]
+
+    if "authentication" not in connection_string.lower():
+        SQL_COPT_SS_ACCESS_TOKEN = 1256
+        attrs_before = {SQL_COPT_SS_ACCESS_TOKEN: db_aad_token_struct()}
+    else:
+        attrs_before = {}
+
+    connection = pyodbc.connect(connection_string, attrs_before=attrs_before)
+    cursor = connection.cursor()
+    logging.info("ODBC connection cursor created.")
+    return cursor
+
+
+def cosmos_client() -> "CosmosClient":
+    """
+    CosmosDB client for connecting with the state store.
+
+    Documentation: https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/sdk-python
+    """
+
+    client = CosmosClient(
+        os.environ["COSMOSDB_ENDPOINT"],
+        credential=(
+            DefaultAzureCredential()
+            if environment != "local"
+            else os.environ["COSMOSDB_KEY"]
+        ),
+        connection_verify=(environment != "local"),
+    )
+    logging.info("Cosmos client created.")
+    return client
+
+
+class CosmosDBLongCallbackManager:
```

Comment:
do we have any plans to clean up expired items?
at present, these will grow over time, won't they?

Reply:
do we have any example code that calls this caching code?

Reply:
* No current clean up code... yes, should add.
* linked above for code from hyacinth